### PR TITLE
Fabric update new canvas, new jsdom

### PR DESCRIPTION
**Pull request by asturur**
- update jsdom to 13
- update canvas to 2
- remove node 4 and 6
- remove xmldom since JSDOM now parses XMLs too
- update test to take care of how JSDOM request resources now
- update visual test loop to create a new canvas and dispose it each test
- update visual test code to do not use the requestRenderCallback
- added NODE 11 to test matrix
- removed allowed failures, since today all test have to pass in node as in browser
- swapped istanbul with nyc
- updated qunit to latest 2.9.2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,8 +60,8 @@
     "testem:ci": "testem ci"
   },
   "optionalDependencies": {
-    "canvas": "^1.6.12",
-    "jsdom": "^9.12.0",
+    "canvas": "^2.0.1",
```
Comment (danielbeardsley) [inline package.json]: Canvas 2.0 has some [breaking changes](https://github.com/Automattic/node-canvas/blob/master/CHANGELOG.md#200)

Comment (asturur) [reply package.json]: i know i know, did not read yet how to interact with new jsdom. Canvas is transparent for us

**Comment by Rosga** [comment]
This is how I use canvas@2.3.1 and jsdom@13.2.0 in my project based an fabric.
```const jsdom = require('jsdom');
const virtualWindow = new jsdom.JSDOM(
  decodeURIComponent('%3C!DOCTYPE%20html%3E%3Chtml%3E%3Chead%3E%3C%2Fhead%3E%3Cbody%3E%3C%2Fbody%3E%3C%2Fhtml%3E'),
  {
    features: {
      FetchExternalResources: ['img']
    },
    resources: "usable"
  }).window;
fabric.document = virtualWindow.document;
fabric.jsdomImplForWrapper = require('jsdom/lib/jsdom/living/generated/utils').implForWrapper;
fabric.nodeCanvas = require('jsdom/lib/jsdom/utils').Canvas;
fabric.window = virtualWindow;
if (typeof DOMParser === "undefined") {
  global.DOMParser = null;
}
DOMParser = require('xmldom').DOMParser;

**Comment by asturur** [comment]
why you do not open a pr, so i can close this?

**Comment by asturur** [comment]
So far so good, we are currently stopped by https://github.com/jsdom/jsdom/pull/2503

If they fix this bug, we can migrate.

**Comment by asturur** [comment]
moved to jsdom 13.1
This could allow us to move forward.

**Comment by asturur** [comment]
ok we have a green build but node tests ( previously marked as possibly fail ) fail harder than before.
I would like to keep the same level of confidence, so just need a little more look.

**Pull request merged**